packages/core/factories/__tests__/useUserAddress.spec.ts

```diff
@@ -0,0 +1,144 @@
+import Vue from 'vue';
+import VueCompositionApi from '@vue/composition-api';
+Vue.use(VueCompositionApi);
+import { useUserAddressFactory, UseUserAddressFactoryParams } from '../src';
+import { UseUserAddress } from '@vue-storefront/interfaces';
+
+let useUserAddress: () => UseUserAddress<any, any>;
+
+let params: UseUserAddressFactoryParams<any, any>;
+let inputAddresses: any = [];
+
+const transform = (addresses: any[]) => ({
+  addresses,
+  updateUserParams: {}
+});
+
+const createComposable = (overrideParams?: UseUserAddressFactoryParams<any, any>) => {
+  params = overrideParams || {
+    userComposable: {
+      updateUser: async () => {}
+    } as any,
+    addresses: inputAddresses,
+    searchAddresses: jest.fn().mockResolvedValueOnce(['address1', 'address2', 'address3']),
+    addAddress: jest.fn().mockResolvedValueOnce(transform(['address1', 'address2', 'address3', 'address4'])),
+    deleteAddress: jest.fn().mockResolvedValueOnce(transform(['address1', 'address2'])),
+    getBillingAddresses: jest.fn().mockReturnValueOnce(['address1']),
+    getShippingAddresses: jest.fn().mockReturnValueOnce(['address2', 'address3']),
+    updateAddress: jest.fn().mockResolvedValueOnce(transform(['address1', 'updatedAddress', 'address3']))
+  };
```

Comment:
I'd suggest this approach:

```suggestion
  params = {
    userComposable: {
      updateUser: async () => {}
    } as any,
    addresses: inputAddresses,
    searchAddresses: jest.fn().mockResolvedValueOnce(['address1', 'address2', 'address3']),
    addAddress: jest.fn().mockResolvedValueOnce(transform(['address1', 'address2', 'address3', 'address4'])),
    deleteAddress: jest.fn().mockResolvedValueOnce(transform(['address1', 'address2'])),
    getBillingAddresses: jest.fn().mockReturnValueOnce(['address1']),
    getShippingAddresses: jest.fn().mockReturnValueOnce(['address2', 'address3']),
    updateAddress: jest.fn().mockResolvedValueOnce(transform(['address1', 'updatedAddress', 'address3'])),
    ...overrideParams
  };
```

then in `overrideParams` you don't have to override all of the params.